### DISCUSSION TITLE
Invalid IR raw command error

### DISCUSSION THREAD

**Discussion by ferbulous** [arendst/Tasmota#21391] [Q&A] [answered]
Hi, my fan has unsupported protocol.
I’m trying to send raw command but I keep getting invalid error

ir signal received
```
04:25:25.315 MQT: tele/tasmota_led/RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":63,"Hash":"0x7BCDA3AE","Repeat":0,"RawData":"+3580-1645+525-335+530-1240+475-390+480-385+500-360I-380IjI-400+505lO-340+520pKjO-135+730-1235GhO-355O-290+575pQpQpCr+90-1010+425-440G-1265GfKhG-295+600tGhImOuImI-1260Z-410OjOp+495jG-1265Z-1315Z-440+450-415KrX-1015+450-1295+445rX-185IjIjI-1265GhImI-1265G-1270+445r+85-1075+450-415+450rX-185+450-415Of+445hK","RawDataInfo":[125,125,0]}}
04:25:25.450 MQT: tele/tasmota_led/RESULT = {"IrReceived":{"Protocol":"UNKNOWN","Bits":23,"Hash":"0x64E6753B","Repeat":0,"RawData":"+360-480+1240-470+1270d+420-445+135-90+185-450+1290-425+1315m+440-475+385k+415g+1295mRkHi+180bSmNmNm+435mOmOmNmOm","RawDataInfo":[46,46,0]}}
```

irsend attempt & error
```
04:34:41.859 CMD: IRSend 0, +3580-1645+525-335+530-1240+475-390+480-385+500-360I-380IjI-400+505lO-340+520pKjO-135+730-1235GhO-355O-290+575pQpQpCr+90-1010+425-440G-1265GfKhG-295+600tGhImOuImI-1260Z-410OjOp+495jG-1265Z-1315Z-440+450-415KrX-1015+450-1295+445rX-185IjIjI-1265GhImI-1265G-1270+445r+85-1075+450-415+450rX-185+450-415Of+445hK
04:34:41.867 MQT: stat/tasmota_led/RESULT = {"IRSend":"Invalid RawData"}
```

What an I missing here and how do I send raw  command when there are 2 lines for each ir signal?

**Comment by NsinghP** [comment, marked as answer]
Hi, I think you have extra space between the 0, and The Raw data string! Try entering this in your console:

`IRSend 0,+3580-1645+525-335+530-1240+475-390+480-385+500-360I-380IjI-400+505lO-340+520pKjO-135+730-1235GhO-355O-290+575pQpQpCr+90-1010+425-440G-1265GfKhG-295+600tGhImOuImI-1260Z-410OjOp+495jG-1265Z-1315Z-440+450-415KrX-1015+450-1295+445rX-185IjIjI-1265GhImI-1265G-1270+445r+85-1075+450-415+450rX-185+450-415Of+445hK`

**Reply by ferbulous** [reply]
Thanks, l just needed to remove the extra space between 0 and the raw data string.
Unfortunately the ceiling fan doesn't seem to receive it, I expected to hear beeping sound but there was nothing.
Or is there a proper way to send the irsend raw data when there's 2 signal lines?
